### PR TITLE
Remove `withRouter` (2/2)

### PR DESCRIPTION
**Pull request by taneliang**
Continues from #3007.

There's 1 remaining `withRouter` callsite in `VenuesContainer`, but I'll fix that in a separate PR as the venues components looks like they're due for an overhaul.

I've fixed the noob React Testing Library mistakes in this PR, but to prevent this PR from ballooning further, I didn't fix the mistakes from #3007. I'll fix them in a follow up PR and add the Testing Library ESLint plugins.

### CODE REVIEW

File: website/src/test-utils/createHistory.ts [hunk 1]
```diff
@@ -11,17 +11,13 @@ type MatchShape = {
   isExact?: boolean;
 };
 
-// This can also be Location, but no test case use that for now so we leave it
```
Comment (taneliang) [inline website/src/test-utils/createHistory.ts:14]: `createMemoryHistory`'s `initialEntries` can only be a `string[]`, so I think this comment is outdated.

File: website/src/views/layout/Navtabs.test.tsx [hunk 2]
```diff
@@ -1,13 +1,71 @@
...
+        "Today",
+        "Timetable",
+        "Modules",
+        "Venues",
+        "Settings",
+        "Contribute",
+        "Whispers",
+      ]
+    `);
+    expect(container).toMatchSnapshot();
```
Comment (taneliang) [inline website/src/views/layout/Navtabs.test.tsx:51]: Not entirely sure how useful these full DOM snapshots are anymore

Comment (taneliang) [reply website/src/views/layout/Navtabs.test.tsx]: Removed

**Review by li-kai — APPROVED** [review]
LGTM 🍎

**Pull request merged**